### ISSUE TITLE
[$60] Issues when creating a challenge

### ISSUE THREAD

**Issue by ThomasKranitsas** [topcoder-platform/challenge-engine-ui#185]
## Steps to reproduce
1. Login as `TonyJ`
2. Select a project (eg `Another Test Project`)
3. Click Launch new to create a new challenge
4. Enter all information
5. Click Launch

## Issues
1. There used to be a `save as draft` button. I'm assuming now we have support for auto-save, this button is hidden as the default challenge status is already draft. I suggest keeping that button and just use it to save and "go back" just in case we don't want to launch right now.
2. Clicking Launch fails with the following error
```error: ErrorCtor [ConflictError]: User TonyJ already has resource with roleId: cfe12b3f-2a24-4639-9d8b-ec86726f76bd in challenge: 288885f8-babf-481c-b19f-e7b68afaecad```

If I then go back, open the challenge from Drafts and then launch it, it's working fine. I guess something is not right in the state management and the UI is trying to create a resource that already exists.

**Comment by ThomasKranitsas** [comment]
cc @jmgasper @rootelement

**Comment by awzilkie** [comment]
Hi @ThomasKranitsas is  this open for pickup?

**Comment by rootelement** [comment]
Yes @awzilkie please see if you can research what's needed and give us a report. We'll price appropriately.

**Comment by awzilkie** [comment]
Ok @rootelement I'll assign it to myself then?

**Comment by Gauravseta** [comment]
@rootelement @ThomasKranitsas will open for pickup label be added? even I was waiting for it

**Comment by rootelement** [comment]
Sorry @Gauravseta , I just got online. There are more issues coming and we will open them for pickup.

**Comment by awzilkie** [comment]
@ThomasKranitsas yes the issue is the autosave with setting of v5 resources. Sometimes while editing the challenge while having set a copilot or reviewer, the autosave creates the resources in the resources-api. Then when trying to launch the resources are already created. The solution is to only update the resources if they have changed since the first autosave.

**Comment by awzilkie** [comment]
FYI @rootelement

**Comment by rootelement** [comment]
Ok please implement that fix @awzilkie

**Comment by awzilkie** [comment]
Ok @rootelement Should I also restore the "save as draft" button as mentioned in issue #1 by @ThomasKranitsas ?

**Comment by jmgasper** [comment]
Contest https://www.topcoder.com/challenges/30123460 has been created for this ticket.<br/><br/>```This is an automated message for ghostar via Topcoder X```

**Comment by jmgasper** [comment]
Contest https://www.topcoder.com/challenges/30123460 has been updated - it has been assigned to azilkie.<br/><br/>```This is an automated message for ghostar via Topcoder X```

**Comment by rootelement** [comment]
> Ok @rootelement Should I also restore the "save as draft" button as mentioned in issue #1 by @ThomasKranitsas ?

Yes, but please just make it trigger the same auto-save functionality, not a separate workflow. Is there a way to implement a faint "last saved" type datetime somewhere on the page? For now, just put it in light gray at the bottom of the page.

**Comment by awzilkie** [comment]
@rootelement currently it is calling the autosave on every input change done by the user. i.e. every keystroke. It would be no problem to add the "last saved" text but it would update on every keystroke. Just FYI

**Comment by rootelement** [comment]
Wow... Every keystroke? Can we debounce that or something? That seems a bit overkill. I'd like to know what @jmgasper thinks before we do that or make that change. Continue with the other fixes.

**Comment by awzilkie** [comment]
oh wait I just noticed that it's throttled to only update every 1s. Sorry :)

**Comment by jmgasper** [comment]
@awzilkie - I'm having trouble reproducing the `error: ErrorCtor [ConflictError]: User TonyJ already has resource with roleId ...` error.  Have you managed to reproduce that?  I just want to make sure we get the exact steps detailed, thanks.

**Comment by awzilkie** [comment]
@jmgasper you have to set a copilot or reviewer to be able to see the error

**Comment by jmgasper** [comment]
@awzilkie - That makes sense, and I can reproduce the issue now, thanks!

**Comment by awzilkie** [comment]
@ThomasKranitsas @jmgasper @rootelement see PR here: https://github.com/topcoder-platform/challenge-engine-ui/pull/186.

Something to note about the autosave: Changes are autosaving after every user input change, but only after the user has inputted enough info to make it a valid challenge. i.e once the 'track', 'typeId', 'name', 'description', 'tags', 'prizeSets' fields are entered. However from the very start a new challenge is created in the API with default values (e.g. "Draft" as the name). So my solution here is to save any changes to the copilot or reviewer to the resources API right from the start so that the Draft created in the API has corresponding resources. Regarding the "Save as Draft" button, I just changed the name and styling of the Cancel button because it already works as a "Save and go Back" (no save needed because of autosave). I would recommend that if we want a "Cancel" button that this will actually delete the challenge that is added by the autosaving.

**Comment by jmgasper** [comment]
@awzilkie - Looks good, thanks.  I'll open up a new ticket for the timestamp display.

**Comment by jmgasper** [comment]
Payment task has been updated: https://software.topcoder.com/review/actions/ViewProjectDetails?pid=30123460<br/><br/>```This is an automated message for ghostar via Topcoder X```
